pyadjoint/reduced_functional.py

```diff
@@ -123,13 +123,15 @@ def derivative(self, adj_input=1.0, options={}):
 
         # Scale adjoint input
         with stop_annotating():
-            adj_value = self.scale * adj_input
+            # Make sure `adj_input` is an OverloadedType
+            adj_input = create_overloaded_object(adj_input)
+            adj_input._ad_mul(self.scale)
```

Comment:
```suggestion
            adj_value = adj_input._ad_mul(self.scale)
```
This operation does not do in-place, it returns the product